### ISSUE TITLE
'watch' task very slow to complete on server reload

### ISSUE THREAD

**Issue by jme783** [ericclemmons/grunt-express-server#83]
Hi there,
I'm seeing some interesting behavior when trying to re-run the `watch` task after the express server stops and starts. On average, it takes 45 seconds for the task to complete before it is able to successfully watch any files that will change. Here's the output from my terminal

```
Running "express:dev" (express) task
Stopping Express server
Starting background Express server
website listening at http://localhost:8080

Running "watch" task
Completed in 47.560s at Thu Mar 19 2015 13:16:31 GMT-0700 (PDT) - Waiting...
```

The times to complete the watch task when first firing up the server, or changing css files are much much lower. Any ideas on what may be causing this?

Here's my `Gruntfile`:
```
module.exports = function(grunt) {

  // Project configuration.
  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json'),
    express: {
      options: {
        port: 8080
      },
      dev: {
        options: {
          script: 'app.js'
        }
      }
    },
    sass: {
      dist: {
        files: [{
          expand: true,
          cwd: 'assets/styles',
          src: ['*.scss'],
          dest: 'public/styles',
          ext: '.css'
        }]
      }
    },
    watch: {
      css: {
        files: '**/*.scss',
        tasks: ['sass'],
        options: {
          livereload: true,
        },
      },
      express: {
        files:  [ '**/*.js', '**/*.hbs' ],
        tasks:  [ 'express:dev' ],
        options: {
          spawn: false // for grunt-contrib-watch v0.5.0+, "nospawn: true" for lower versions. Without this option specified express won't be reloaded
        }
      }
    }

  });

  // Load the plugin that provides the "uglify" task.
  grunt.loadNpmTasks('grunt-contrib-sass');
  grunt.loadNpmTasks('grunt-contrib-watch');
  grunt.loadNpmTasks('grunt-express-server');

  // Default task(s).
  grunt.registerTask('server', ['express:dev', 'watch']);

};
```

and my `app.js`
```
var express = require('express');
var path = require('path');
var favicon = require('serve-favicon');
var logger = require('morgan');
var cookieParser = require('cookie-parser');
var bodyParser = require('body-parser');
var exphbs  = require('express-handlebars');

var app = express();
var routes = require('./routes')(app);
// view engine setup
app.engine('.hbs', exphbs({defaultLayout: 'main', extname: '.hbs'}));
app.set('view engine', '.hbs');

// uncomment after placing your favicon in /public
//app.use(favicon(__dirname + '/public/favicon.ico'));
app.use(logger('dev'));
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: false }));
app.use(cookieParser());

app.use(express.static(path.join(__dirname, 'public')));
// catch 404 and forward to error handler
app.use(function(req, res, next) {
  var err = new Error('Not Found');
  err.status = 404;
  next(err);
});

// error handlers

// development error handler
// will print stacktrace
if (app.get('env') === 'development') {
  app.use(function(err, req, res, next) {
    res.status(err.status || 500);
    res.render('error', {
      message: err.message,
      error: err
    });
  });
}

// production error handler
// no stacktraces leaked to user
app.use(function(err, req, res, next) {
  res.status(err.status || 500);
  res.render('error', {
    message: err.message,
    error: {}
  });
});

// Server
var server = app.listen(8080, function() {
  var host = server.address().address;
  var port = server.address().port;
  console.log('website listening at http://%s:%s', host, port);
});

module.exports = app;
```

**Comment by zelias500** [comment]
I just encountered this problem -- the watch task takes so long because you are watching a ton of files. Loads quickly the first time because the watch task doesn't need to check each file that is being watched on initial load. 
You currently have: 
`watch: {
     express: {
        files:  [ '**/*.js', '**/*.hbs' ],
       // ...
        }
      }`

This means you're watching every single JS file in your directory, including any 3rd party libraries that themselves may be really large. I cut down my watch task from ~30s down to ~2s by only watching my own code:
`watch: {
   express: {
      files: ['source/js/**/*.js'
       // ...
    }
}`

Not an optimal solution, but I found this got my load time down.
